Frontend serving redesign

VoyTechnology opened the pull request:
This redesign changes how the frontend is handled. It moves the frontend
of the show serving code into its internal/frontend package, which will
contain all components of serving the frontend.

The new server package contains a new instance-based server, which can
be extended using components which are served on the path-prefix. This
Also removes the code responsible for loading individual hosts.

Commit:
Frontend serving redesign

This redesign changes how the frontend is handled. It moves the frontend
of the show serving code into its internal/frontend package, which will
contain all components of serving the frontend.

The new server package contains a new instance-based server, which can
be extended using components which are served on the path-prefix. This
Also removes the code responsible for loading individual hosts.

RyanConnell commented on cmd/frontend/frontend.go:
```diff
@@ -17,22 +24,51 @@ func main() {
+
+// Config used for the frontend.
+// TODO: Add default values once we deprecate file settings.
+type Config struct {
```
nit: can we keep the struct and it's methods at the top of the file (under imports ofc); all other methods can go beneath the struct methods

RyanConnell commented on internal/frontend/show.go:
```diff
@@ -261,3 +246,28 @@ func (f *Frontend) get(ctx context.Context, url string, v interface{}) error {
 }
+
+// Option allows to modify the frontend.
+type Option func(*ShowFrontend) error
```
In the future will this be renamed to something like 'ShowOption' so that we can support having multiple of these type aliases? (i.e., one for ArtistOption, MovieOption, etc)

VoyTechnology committed:
Renamed types and changed order in cmd/frontend

internal/frontend.Option was renamed to ShowOption, as this will have
future name collisions.

VoyTechnology committed:
Renamed internal/server to internal/httpserver

Addition of the protocol is a better package name than if it was not
included.
